### ISSUE TITLE
App outputs "G@" on quit

### ISSUE THREAD

**Issue by jiri** [coreyoconnor/vty#89]
When I write a dead simple vty app, akin to

```haskell
main = do
  vty <- mkVty def
  _ <- liftIO (nextEvent vty)
  shutdown vty
```

I get the characters `g@` output when it quits. I suspect it's actually something else, possibly binary data being outputted, because when I use another terminal app, it messes up my prompt permanently for that session.

I'm using OS X with Terminal and iTerm2.

**Comment by jiri** [comment]
Upon further investigation, nothing is outputted when using `bash` as my shell (I use `fish`). This leads me to believe that the characters outputted are actually a part of the prompt that gets messed up, which would indicate that `vty` messes up the encoding / locale.

This is a bit more serious issue than I initially thought, so I'd like to see some progress on it. I'll take a look at the source code and see if I can come up with a pull request :)

**Comment by jtdaugherty** [comment]
I'm encountering this behavior with the latest Vty only when using OS X's Terminal.app.  For me, the output is `G@` just before the shell prompt after exiting my Vty program.  This looks to be related to the UTF8 charset management code that sends escapes containing `G` and `@`.  This behavior doesn't show up on iTerm2, so there's that.

**Comment by jtdaugherty** [comment]
For what it's worth, `tmux`'s behavior is to check `LC_*` and `LANG` environment variables for the strings `utf-8` and `utf8` to determine whether UTF-8 is supported.  Terminal.app sets `LANG` accordingly on my system (which is a consequence of the default Terminal.app preferences).  So one alternative to always emitting those sequences would be to only emit them when the environment *lacks* a UTF-8 declaration.

**Comment by glguy** [comment]
@jtdaugherty  the behavior you describe sounds reasonable. Also, vty has in the past detected the TERM_PROGRAM environment var terminal.app sets and used that to enable compatibility behaviors.

**Comment by jtdaugherty** [comment]
This fix has been released in 5.9.1.

**Comment by jiri** [comment]
Amazing, thank you!
